### PR TITLE
UCP: fix an error resulting from sending a small message with rndv.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

File: src/ucp/tag/rndv.c [hunk 1]
```diff
@@ -125,6 +125,9 @@ ucs_status_t ucp_proto_progress_rndv_get(uct_pending_req_t *self)
 
     if ((offset == 0) && ((uintptr_t)get_req->send.buffer % UCP_ALIGN)) {
         length = UCP_MTU_SIZE - ((uintptr_t)get_req->send.buffer % UCP_ALIGN);
+        if (length > get_req->send.length) {
```
Comment (yosefe) [inline src/ucp/tag/rndv.c]: maybe better to check if length>MTU in line 126?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2314/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1639/ for details (Mellanox internal link).

File: src/ucp/tag/rndv.c [hunk 2]
```diff
@@ -123,11 +123,9 @@ ucs_status_t ucp_proto_progress_rndv_get(uct_pending_req_t *self)
 
     offset = get_req->send.state.offset;
 
-    if ((offset == 0) && ((uintptr_t)get_req->send.buffer % UCP_ALIGN)) {
+    if ((offset == 0) && ((uintptr_t)get_req->send.buffer % UCP_ALIGN) &&
+        (get_req->send.length > (UCP_MTU_SIZE - ((uintptr_t)get_req->send.buffer % UCP_ALIGN)))) {
```
Comment (yosefe) [inline src/ucp/tag/rndv.c]: why not just get_req->send.length > UCP_MTU_SIZE?

Comment (alinask) [inline src/ucp/tag/rndv.c]: made it the same as it was in the prevous commit.
i'll remove.

**Comment by alinask** [comment]
there is a hang if it's removed.

File: src/ucp/tag/rndv.c (hunk 2 above)
Comment (alinask) [inline src/ucp/tag/rndv.c]: there is a hang if it's removed.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2316/ for details.

File: src/ucp/tag/rndv.c (hunk 2 above)
Comment (shamisp) [inline src/ucp/tag/rndv.c]: What is UCP_ALIGN 256 ?  header space ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1641/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2317/ for details.

**Comment by yosefe** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1642/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls review

**Comment by shamisp** [comment]
👍

**Pull request merged**